pkg/image/api/helper.go

```diff
@@ -184,7 +184,41 @@ func ParseDockerImageReference(spec string) (DockerImageReference, error) {
 		ref.ID = id
 		break
 	default:
-		// TODO: this is no longer true with V2
```

Comment:
Regarding `case 3`: I feel like it's about time to recognize `registry` part based on special characters it contains. The same way [upstream does it](https://github.com/docker/docker/blob/0cc7f0185f59f9271b7b783b776dac74347e8f51/reference/reference.go#L176).